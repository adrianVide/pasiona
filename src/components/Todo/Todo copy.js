import React from "react";
import axios from "axios";
import { useEffect, useState } from "react";
import SingleTodo from "../SingleTodo/SingleTodo.js";

const Todo = () => {
  const [todos, setTodos] = useState([]);
  

  useEffect(async () => {
    const result = await axios("http://localhost:3000/todos");
    setTodos(result.data);
  }, []);

  return (
    <div>
      {todos.map((todo) => (
        <SingleTodo todo={todo} />
      ))}
    </div>
  );
};

export default Todo;
