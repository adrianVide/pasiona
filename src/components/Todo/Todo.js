import React from "react";
import axios from "axios";
import { useEffect, useState } from "react";
import SingleTodo from "../SingleTodo/SingleTodo.js";
import AddTodo from "../AddTodo/AddTodo.js";

const Todo = () => {
  const [todos, setTodos] = useState([]);
  const [reload, setReload] = useState(false);

  useEffect(async () => {
    try {
      const result = await axios("http://localhost:3000/todos");
      setTodos(result.data);
    } catch (error) {
      console.log(error);
    }
  }, [reload]);

  return (
    <>
      <AddTodo callBack={() => setReload(!reload)} />

      {todos
        .map((todo) => (
          <SingleTodo todo={todo} callBack={() => setReload(!reload)} />
        ))
        .reverse()}
    </>
  );
};

export default Todo;
