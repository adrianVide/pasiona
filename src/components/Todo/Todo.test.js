import React from "react";
import ReactDOM from "react-dom";
import Todo from "./Todo";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer'



afterEach(cleanup)


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ul></ul>, div);
});

it("renders trigger correctly", () => {
  const { getByTestId } = render(<Todo />);
  expect(getByTestId("list-group")).toContainHTML('<ul') 
});

it("matches snapshot", () => {
  const compTree = renderer
    .create(<Todo />)
    .toJSON();
  expect(compTree).toMatchSnapshot();
});