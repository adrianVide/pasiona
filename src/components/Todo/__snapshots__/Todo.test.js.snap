// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="d-flex justify-content-center bg-gray-100 rounded-xl py-6"
    data-testid="add-todo"
  >
    <form
      className="d-flex flex-column justify-content-center text-center align-items-center"
      onSubmit={[Function]}
    >
      <h1
        className="text-gray-800 font-bold "
      >
        Add a new Task
      </h1>
      <input
        className="shadow appearance-none border rounded py-2 px-3 text-grey-darker m-1"
        name="title"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value=""
      />
      <input
        className="shadow appearance-none border rounded py-2 px-3 text-grey-darker m-1"
        name="body"
        onChange={[Function]}
        placeholder="Body"
        type="text"
        value=""
      />
      <input
        className="shadow appearance-none border rounded py-2 px-6 text-grey-darker m-1"
        type="submit"
        value="Add Task"
      />
    </form>
  </div>,
  <ul
    data-testid="list-group"
  />,
]
`;
